### ISSUE TITLE
Add IRIX file monitoring to glib2

### ISSUE THREAD

**Issue by danielhams** [sgidevnet/sgug-rse#155]
IRIX has a set of APIs for filesystem monitoring.

glib2 currently in missing implementation of file monitoring - and it would benefit at least one application sat on top (microdnf).

example: build a little C program that uses glib2 file monitoring - and try to create a file monitor on a directory -> will show the problem.

**Comment by danielhams** [comment]
Deeper info - Not working directory monitoring:

```
[sgugshell dan@leafy glibfilemonitoring]$ pwd
/usr/people/dan/Temp/glibfilemonitoring
[sgugshell dan@leafy glibfilemonitoring]$ ./buildit.sh
[sgugshell dan@leafy glibfilemonitoring]$ G_MESSAGES_DEUBG=1 ./g_file_monitor /tmp/
unable to monitor /tmp/: Unable to find default local file monitor type
[sgugshell dan@leafy glibfilemonitoring]$ 
```

So above you can see that directory monitoring doesn't work.

Working(ish) file monitoring (it is polling):

```
[sgugshell dan@leafy glibfilemonitoring]$ G_MESSAGES_DEBUG=1 ./g_file_monitor /tmp/test1.txt &
[1] 1433948
[sgugshell dan@leafy glibfilemonitoring]$ monitoring /tmp/test1.txt

[sgugshell dan@leafy glibfilemonitoring]$ 
[sgugshell dan@leafy glibfilemonitoring]$ touch /tmp/test1.txt
[sgugshell dan@leafy glibfilemonitoring]$ /tmp/test1.txt contents changed
/tmp/test1.txt set of changes done

[sgugshell dan@leafy glibfilemonitoring]$ rm /tmp/test1.txt
[sgugshell dan@leafy glibfilemonitoring]$ /tmp/test1.txt deleted

[sgugshell dan@leafy glibfilemonitoring]$ touch /tmp/test1.txt
[sgugshell dan@leafy glibfilemonitoring]$ /tmp/test1.txt created
/tmp/test1.txt set of changes done

fg
G_MESSAGES_DEBUG=1 ./g_file_monitor /tmp/test1.txt
(I CTRL^C it at this point)
```
Example program take from here: https://gist.githubusercontent.com/simu/0ed1e9b12871e20cec48efe73eeeb8e1/raw/c4cec27552b01dd868610cdf3c9579e44384c29b/g_file_monitor.c

**Comment by danielhams** [comment]
Fixed with c960e5fc10945b9279dd33ad4c85a076bc47ad4b, will be in 0.0.7
